proxy/client/validator/validator.go

```diff
@@ -1,16 +1,300 @@
+	if err != nil {
+		panic(err)
+	}
+
+	outputFacetSchemas, err := compileOutputFacetSchemas(compiler)
+	if err != nil {
+		panic(err)
+	}
+
+	baseDatasetFacetSchema, err := compiler.Compile("https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/DatasetFacet")
+	if err != nil {
+		panic(err)
+	}
+
+	datasetFacetSchemas, err := compileDatasetFacetSchemas(compiler)
+	if err != nil {
+		panic(err)
+	}
+
+	return &validator{
+		mainSchema:             mainSchema,
+		baseRunFacetSchema:     baseRunFacetSchema,
+		baseJobFacetSchema:     baseJobFacetSchema,
+		runFacetSchemas:        runFacetSchemas,
+		jobFacetSchemas:        jobFacetSchemas,
+		baseInputFacetSchema:   baseInputFacetSchema,
+		inputFacetSchemas:      inputFacetSchemas,
+		baseOutputFacetSchema:  baseOutputFacetSchema,
+		outputFacetSchemas:     outputFacetSchemas,
+		baseDatasetFacetSchema: baseDatasetFacetSchema,
+		datasetFacetSchemas:    datasetFacetSchemas,
+	}
+}
+
+func compileRunFacetSchemas(compiler *jsonschema.Compiler) (schemas []*jsonschema.Schema, err error) {
+	facets := []string{
+		"ErrorMessageRunFacet.json",
+		"ExternalQueryRunFacet.json",
+		"ExtractionErrorRunFacet.json",
+		"NominalTimeRunFacet.json",
+		"ParentRunFacet.json",
+		"ProcessingEngineRunFacet.json",
+	}
+	return compileFacetSchemas(compiler, facets)
+}
+
+func compileJobFacetSchemas(compiler *jsonschema.Compiler) (schemas []*jsonschema.Schema, err error) {
+	facets := []string{
+		"DocumentationJobFacet.json",
+		"OwnershipJobFacet.json",
+		"SourceCodeJobFacet.json",
+		"SourceCodeLocationJobFacet.json",
+		"SQLJobFacet.json",
+	}
+	return compileFacetSchemas(compiler, facets)
+}
+
+func compileInputFacetSchemas(compiler *jsonschema.Compiler) (schemas []*jsonschema.Schema, err error) {
+	facets := []string{"DataQualityMetricsInputDatasetFacet.json"}
+	return compileFacetSchemas(compiler, facets)
+}
+
+func compileOutputFacetSchemas(compiler *jsonschema.Compiler) (schemas []*jsonschema.Schema, err error) {
+	facets := []string{"OutputStatisticsOutputDatasetFacet.json"}
+	return compileFacetSchemas(compiler, facets)
+}
+
+func compileDatasetFacetSchemas(compiler *jsonschema.Compiler) (schemas []*jsonschema.Schema, err error) {
+	facets := []string{
+		"ColumnLineageDatasetFacet.json",
+		"DatasetVersionDatasetFacet.json",
+		"DatasourceDatasetFacet.json",
+		"DataQualityAssertionsDatasetFacet.json",
+		"DocumentationDatasetFacet.json",
+		"LifecycleStateChangeDatasetFacet.json",
+		"OwnershipDatasetFacet.json",
+		"SchemaDatasetFacet.json",
+		"StorageDatasetFacet.json",
+		"SymlinksDatasetFacet.json",
+	}
+	return compileFacetSchemas(compiler, facets)
+}
+
+func compileFacetSchemas(compiler *jsonschema.Compiler, facets []string) (schemas []*jsonschema.Schema, err error) {
+	schemas = make([]*jsonschema.Schema, len(facets))
+	for i, schemaKey := range facets {
+		schemaURL := schemaURLs[schemaKey]
+		schemas[i], err = compiler.Compile(schemaURL)
+		if err != nil {
+			return
```

Comment:
I know this is legal, but I find it jarring. I prefer always specifying return variables rather than just relying on the return signature (I might call this a medium-sized function in reference to https://google.github.io/styleguide/go/decisions#named-result-parameters and its advice re: Naked returns)